integration-tests/tests/api/schema/check.spec.ts

```diff
@@ -1885,3 +1885,359 @@ test.concurrent(
+        directive @inputFieldDefinitionCustomDirectiveTest on INPUT_FIELD_DEFINITION
+        directive @unionCustomDirectiveTest on UNION
+
+        type Query {
+          a: String
+        }
+
+        type ObjectTest @objectCustomDirectiveTest {
+          a: String
+        }
+
+        scalar NewScalar @scalarCustomDirectiveTest
+
+        interface InterfaceTest @interfaceCustomDirectiveTest {
+          a: String
+        }
+
+        input InputTest @inputObjectCustomDirectiveTest {
+          a: String
+          b: String @inputFieldDefinitionCustomDirectiveTest
+        }
+
+        type ArgumentDefinitionTest {
+          a(a: String @argumentDefinitionCustomDirectiveTest): String
+        }
+
+        enum EnumTest @enumCustomDirectiveTest {
+          A
+          B @enumValueCustomDirectiveTest
+        }
+
+        type A {
+          a: String!
+        }
+
+        type B {
+          b: String!
+        }
+
+        union UnionTest @unionCustomDirectiveTest = A | B
+      `,
+    })
+    .then(r => r.expectNoGraphQLErrors());
+
+  // Schema publish should be successful
+  expect(publishResult.schemaPublish.__typename).toBe('SchemaPublishSuccess');
+
+  // Create a token with read rights
+  const readToken = await createTargetAccessToken({
+    mode: 'readOnly',
+  });
+
+  // Check schema with removed custom directives
+  const checkResult = await readToken
+    .checkSchema(/* GraphQL */ `
+      type Query {
+        ping: String
+      }
+    `)
+    .then(r => r.expectNoGraphQLErrors());
+
+  const schemaCheck = checkResult.schemaCheck;
+
+  if (schemaCheck.__typename !== 'SchemaCheckError') {
+    throw new Error(`Expected SchemaCheckError, got ${schemaCheck.__typename}`);
+  }
+
+  const schemaCheckId = schemaCheck.schemaCheck?.id;
+
+  if (schemaCheckId == null) {
+    throw new Error('Missing schema check id.');
+  }
+
+  expect(schemaCheck.changes?.total).toEqual(21);
+  expect(schemaCheck.changes?.nodes).toEqual([
+    { message: "Type 'A' was removed", criticality: 'Breaking' },
+    {
+      message: "Type 'ArgumentDefinitionTest' was removed",
+      criticality: 'Breaking',
+    },
+    { message: "Type 'B' was removed", criticality: 'Breaking' },
+    { message: "Type 'EnumTest' was removed", criticality: 'Breaking' },
+    { message: "Type 'InputTest' was removed", criticality: 'Breaking' },
+    {
+      message: "Type 'InterfaceTest' was removed",
+      criticality: 'Breaking',
+    },
+    { message: "Type 'NewScalar' was removed", criticality: 'Breaking' },
+    { message: "Type 'ObjectTest' was removed", criticality: 'Breaking' },
+    { message: "Type 'UnionTest' was removed", criticality: 'Breaking' },
+    {
+      message: "Field 'a' was removed from object type 'Query'",
+      criticality: 'Breaking',
+    },
+    {
+      message: "Directive 'argumentDefinitionCustomDirectiveTest' was removed",
+      criticality: 'Breaking',
+    },
```

Comment:
Why do we treat removal of a directive as breaking?

Reply:
it will confuse the schema check, as the directive is never used.
A graphql query does not include the directive, it's only available on the schema.

Reply:
I think everything related to directives, should be a warning